pkg/simulation/config/envoyfilter.go

```diff
@@ -0,0 +1,98 @@
+package config
+
+import (
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	"istio.io/client-go/pkg/apis/networking/v1alpha3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/howardjohn/pilot-load/pkg/simulation/model"
+)
+
+type EnvoyFilterSpec struct {
+	App            string
+	Namespace      string
+	connectTimeout int
+	Parent         model.IstioAPIParent
+}
+
+type EnvoyFilter struct {
+	Spec *EnvoyFilterSpec
+}
+
+var _ model.Simulation = &EnvoyFilter{}
+
+func NewEnvoyFilter(s EnvoyFilterSpec) *EnvoyFilter {
+	return &EnvoyFilter{Spec: &s}
+}
+
+func (v *EnvoyFilter) Refresh(ctx model.Context) error {
+	v.Spec.connectTimeout = (v.Spec.connectTimeout + 1) % 10
+	return v.Run(ctx)
+}
+
+func (v *EnvoyFilter) Run(ctx model.Context) (err error) {
+	return ctx.Client.Apply(v.getEnvoyFilter())
+}
+
+func (v *EnvoyFilter) Cleanup(ctx model.Context) error {
+	return ctx.Client.Delete(v.getEnvoyFilter())
+}
+
+func (v *EnvoyFilter) getEnvoyFilter() *v1alpha3.EnvoyFilter {
+	s := v.Spec
+	spec := networkingv1alpha3.EnvoyFilter{}
+
+	name := s.Namespace
+	applyTo := networkingv1alpha3.EnvoyFilter_NETWORK_FILTER
```

Comment:
do all of these configs we are setting actually work? I am surprised the same patch works with all these different applyTo and does not generate invalid config

Reply:
Yeah, they might conflict with each other, I can have a single config for all levels (root ns, ns and application level) but wondering refresh capability might introduce conflict as config will be changed for application level in refresh operation.

Reply:
Were these actualyl tested? I am worried these will be completely invalid and not accepted by Istiod

Reply:
Now I implemented a trace sampling envoyFilter and tested that it is being created without any error logs in istiod. Please check and merge the PR if looks good. I don't have permissions to merge the PR.